app/models/edition/identifiable.rb

```diff
@@ -16,10 +16,15 @@ def linkable?
   end
 
   def ensure_presence_of_document
-    self.document ||= Document.new(
-      sluggable_string: string_for_slug,
-      content_id: SecureRandom.uuid,
-    )
+    if document.blank?
+      self.document = Document.new(
+        sluggable_string: string_for_slug,
+        content_id: SecureRandom.uuid,
+      )
+    elsif document.new_record?
+      document.sluggable_string = string_for_slug if document.slug.blank?
+      document.content_id = SecureRandom.uuid if document.content_id.blank?
+    end
```

Comment:
Yea i tried that it and it still didn't work. I think there's something a bit strange going on deep down in the code somewhere. 

Without the conditionals it broke some tests. It might be that the tests can be removed/are out of date, but i didn't really want to go down a rabbit hole as we had limited time.